options.go

```diff
@@ -31,3 +41,52 @@ func withSkipWhitespace() Option {
 		return nil
 	}
 }
+
+// WithColumnMap provides an optional map of columns from a column in the user
+// provided query to a column in the database model
+func WithColumnMap(m map[string]string) Option {
+	return func(o *options) error {
+		if !isNil(m) {
+			o.withColumnMap = m
+		}
+		return nil
+	}
+}
+
+// ValidateConvertFunc validates the value and then converts the columnName,
+// comparisonOp and value to a WhereClause
+type ValidateConvertFunc func(columnName string, comparisonOp ComparisonOp, value *string) (*WhereClause, error)
+
+// WithConverter provides an optional ConvertFunc for a column identifier in the
+// query.This allows you to provide whatever custom validation+conversion you
+// need on a per column basis.  See: DefaultValidateConvert(...) for inspiration.
+func WithConverter(fieldName string, fn ValidateConvertFunc) Option {
+	const op = "mql.WithSqlConverter"
+	return func(o *options) error {
+		switch {
+		case fieldName != "" && !isNil(fn):
+			o.withValidateConvertFn = fn
+			o.withValidateConvertColumn = fieldName
+		case fieldName == "" && !isNil(fn):
+			return fmt.Errorf("%s: missing field name: %w", op, ErrInvalidParameter)
+		case fieldName != "" && isNil(fn):
+			return fmt.Errorf("%s: missing ConvertToSqlFunc: %w", op, ErrInvalidParameter)
+		}
+		return nil
+	}
+}
+
+// WithIgnoredFields provides an optional list of fields to ignore in the model
+// when parsing. Note: Field names are case sensitive.
```

Comment:
Worth saying whether this is Go field names or SQL column names.

Reply:
Isn't that state with in "fields to ignore in the model"?

Reply:
I thought it was a bit ambiguous, I agree fields should really not leave much room for interpretation, but maybe say `struct fields` to make it even more clear?

Reply:
better?